public/script.js

```diff
@@ -357,12 +357,20 @@ function getState(map, projection) {
       let totalDays = Math.abs(firstDate.diff(lastDate, 'days') + 1);
       let forwardDays = (totalDays / 4) * i;
       let scaleDate = moment(firstDate).add(forwardDays, 'd');
-      $(`#scale${i}`).html(scaleDate.format('YYYY'));
+      if (STATE.temporality == 'monthly') {
+        $(`#scale${i}`).html(scaleDate.format('YYYY-MM'));
+      } else {
+        $(`#scale${i}`).html(scaleDate.format('YYYY-MM-DD'));
+      }
     } else {
       let totalDays = Math.abs(STATE.start.diff(STATE.end, 'days') + 1);
       let forwardDays = (totalDays / 4) * i;
       let scaleDate = moment(STATE.start).add(forwardDays, 'd');
-      $(`#scale${i}`).html(scaleDate.format('YYYY'));
+      if (STATE.temporality == 'monthly') {
```

Comment:
See above comment...and consider making a helper function to do some of this work!